### PR TITLE
FindBugs and PMD appeased

### PR DESCRIPTION
**Pull request by iskradelta**
Starter issue. Working on satisfying PMD next.

### CODE REVIEW

**Comment by iskradelta** [comment]
This is for PR #322

**Comment by AndyScherzinger** [comment]
Hi @iskradelta thank you very much for this PR! :tada: - @tobiasKaminsky can you have a look/test drive since I am currently away from my dev maschine for the next couple of days...

cc @djubreel since you are also working on findbug/pmd/etc.

corresponding issue: #322

**Comment by iskradelta** [comment]
Now PMD is appeased, squashed many warnings/errors, there was about 190-200ish at start, and about 20-30ish were suppressed by annotation, mostly duplicate string literals which would cause the code to be unreadable, and src/third_party added to exclude pattern.

What else needs to be done? Do we have common code formatting checks in CI?

**Comment by tobiasKaminsky** [comment]
Great work! I will have look tomorrow.

> Do we have common code formatting checks in CI?

Currently not. We want to switch to the "default" code formatting found in AndroidStudio with line break at 120. (https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#android-studio-formatter-setup)
If you know a good tool for this, please open a new issue and we can discuss it there.
Really appreciate all your work!

File: src/com/owncloud/android/MainApp.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ public static void showOnlyFilesOnDevice(boolean state){
         mOnlyOnDevice = state;
     }
 
-    public static boolean getOnlyOnDevice(){
+    public static boolean isnlyOnDevice(){
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/MainApp.java]: typo --> isOnlyOnDevice

File: src/com/owncloud/android/operations/UploadFileOperation.java [hunk 2]
```diff
@@ -760,8 +765,10 @@ private void move(File sourceFile, File targetFile) throws IOException {
                         // the best option could be show a warning message
                     }
                     finally {
-                        if (inChannel != null) inChannel.close();
-                        if (outChannel != null) outChannel.close();
+                        if (inChannel != null) {
+                            inChannel.close();
+                        }
+                        outChannel.close();
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/operations/UploadFileOperation.java]: shouldn't the null check be kept for outChannel also?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky please also have a look as like yo mentioned. I did and am fine with the changes, will have to run a test though. Other than that :+1:

**Comment by tobiasKaminsky** [comment]
Looks good.
I will include it first in beta for some time.
We will merge it very close to end of october to get your tshirt ;-)

**Comment by AndyScherzinger** [comment]
Sounds good to me! :+1:

**Comment by AndyScherzinger** [comment]
The branch has conflicts Btw :(

**Comment by AndyScherzinger** [comment]
@iskradelta can you resolve the conflicts?
@tobiasKaminsky we are close to November, so this should get merged very soon.

**Comment by iskradelta** [comment]
Sorry a bit late response.

Uh, a little help here, Ive resolved the conflicts, rebased these commits ontop of master, and force pushed to iskradelta/android.git but it doesnt update this pull-request here? Or does it?

**Comment by AndyScherzinger** [comment]
it does :) :+1: PRs are automatically changes since the originating branch got the changes :smiley:

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky did you test the PR? My test was negative (file listing always empty on any account on any server) on my phone but I will do another test after cleaning the installation (jumping back an d forth between DB versions, so so there are different aspects that might have an impact here).

**Comment by tobiasKaminsky** [comment]
Strange, for me it is also not working on a fresh clean emulator :/

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I found the empty list issue a ```!``` got lost when the pmd/findbugs/etc. demanded changes were done. This works now on my phone but I haven't made a larger test covering the different areas of the app, just to make sure we don't have any other issues.

**Comment by AndyScherzinger** [comment]
@iskradelta @tobiasKaminsky I found another issue. Navigation via the drawer menu from "Uploads" to "Settings" crashes the app with a NPE.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I'd say id is ready for beta (could you merge, or should I) but should survive there first before we merge it to master.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger please do it. I am currently busy ;-) ;-)

**Comment by AndyScherzinger** [comment]
Imho ready to merge @tobiasKaminsky :+1:

**Comment by AndyScherzinger** [comment]
Hi @iskradelta,

Thanks for yet another great PR pushing the code quality of the app! :tada:

**Pull request merged**